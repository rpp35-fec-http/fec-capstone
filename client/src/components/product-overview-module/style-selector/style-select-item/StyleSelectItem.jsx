import React, { useState } from 'react';
import './StyleSelectItem.scss';
import CheckIcon from '@mui/icons-material/Check';
import Avatar from '@mui/material/Avatar';
import DefaultImage from '../../../pages/home/product-card/Image-coming-soon.svg';

const Style = ({ style_id, image, handleStyleClick, currentStyle, index }) => {
   const [active, setActive] = useState(false);
   const handleClick = () => {
      handleStyleClick(style_id);
   };

   return (
      <div data-testid='style-select-item' onClick={handleClick}>
         {currentStyle.style_id === style_id ? (
            <>
               <Avatar
                  className='style-item-icon'
                  sx={{
                     height: '25px',
                     width: '25px',
                     position: 'absolute',
                     zIndex: '2',
                     bgcolor: '#1976D2',
                  }}
               >
                  <CheckIcon />
               </Avatar>
               <img
                  className='style-item-active-image'
                  alt='active thumbnail image'
                  src={image || DefaultImage}
               ></img>
            </>
         ) : (
            <img
               className='style-item-inactive-image'
               src={image || DefaultImage}
               alt='style image not found'
               data-testid={`style-img-${index}`}
            ></img>
         )}
      </div>
   );
};

export default Style;
