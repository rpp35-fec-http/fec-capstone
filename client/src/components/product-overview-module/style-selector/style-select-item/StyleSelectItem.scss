@use '../../../App.scss' as *;

// .style-item-active-image {
//   @include set-size(auto, 10em);
// }

.style-item-inactive-image {
  @include set-size(auto, 10em);
}
