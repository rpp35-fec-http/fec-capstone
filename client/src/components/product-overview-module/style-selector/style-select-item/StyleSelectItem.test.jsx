import renderer from 'react-test-renderer';
import StyleSelectItem from './StyleSelectItem.jsx';
import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import 'regenerator-runtime/runtime';

const currentStyle = { style_id: 1243 };

test('renders correctly', () => {
   const tree = renderer
      .create(<StyleSelectItem currentStyle={currentStyle} />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('displays the image without checkbox when not selected', () => {
   const user = userEvent.setup();
   render(
      <StyleSelectItem
         style_id={11}
         image={'test'}
         handleStyleClick={'test'}
         index={0}
         currentStyle={currentStyle}
      />
   );
   expect(screen.getByTestId('style-img-0')).toBeDefined();
});

test('displays the checkbox when selected', () => {
   const user = userEvent.setup();
   render(
      <StyleSelectItem
         style_id={1243}
         image={'test'}
         handleStyleClick={'test'}
         index={0}
         currentStyle={currentStyle}
      />
   );

   expect(screen.getByTestId('checkbox-icon')).toBeDefined();
});
