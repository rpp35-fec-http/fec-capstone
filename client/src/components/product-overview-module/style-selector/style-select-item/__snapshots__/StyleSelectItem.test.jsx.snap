// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="style-select-item"
  onClick={[Function]}
>
  <img
    alt="style image not found"
    className="style-item-inactive-image"
    data-testid="style-img-undefined"
    src="test-file-stub"
  />
</div>
`;
